src/internal/storage/fileset/index/reader.go

```diff
@@ -58,23 +90,42 @@ func (r *Reader) Iterate(ctx context.Context, cb func(*Index) error) error {
 	}
 	traverseCb := func(idx *Index) (bool, error) {
 		if atEnd(idx.Path, r.filter) {
+			r.skippedIdx = append(r.skippedIdx, toEdge(r.prevIdx, idx, "past end"))
 			return false, errutil.ErrBreak
 		}
 		if idx.File != nil {
 			if !atStart(idx.Path, r.filter) || !(r.datum == "" || r.datum == idx.File.Datum) {
+				r.skippedIdx = append(r.skippedIdx, toEdge(r.prevIdx, idx, "file before start / datum mismatch"))
 				return false, nil
 			}
+			r.traversedIdx = append(r.traversedIdx, toEdge(r.prevIdx, idx, "traversed file"))
 			return false, cb(idx)
 		}
 		if !atStart(idx.Range.LastPath, r.filter) {
+			r.skippedIdx = append(r.skippedIdx, toEdge(r.prevIdx, idx, "range before start"))
 			return false, nil
 		}
+		r.traversedIdx = append(r.traversedIdx, toEdge(r.prevIdx, idx, "traversed range"))
 		return true, nil
 	}
 	_, err := r.traverse(ctx, r.topIdx, []byte{}, traverseCb)
 	if errors.Is(err, errutil.ErrBreak) {
 		err = nil
 	}
+	traversedEdges := ""
+	for _, edge := range r.traversedIdx {
+		traversedEdges += edge.String() + "\n"
+	}
+	skippedEdges := ""
+	for _, edge := range r.skippedIdx {
+		skippedEdges += edge.String() + "\n"
+	}
+	if len(skippedEdges)+len(traversedEdges) > 0 {
+		fmt.Println("traversed indices:", len(r.traversedIdx), "skipped indices:", len(r.skippedIdx))
```

Comment:
This should be `log.Info(pctx.TODO(), ...)`.  This messes up anything else that relies on the logs being JSON.